client/src/components/WaitingRoom/index.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { compose } from 'recompose';
+import { withAuthorization } from '../Session';
+
+class WaitingRoomBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      players: [],
+      gameId: '',
+      timeLimit: 10,
+      started: false
+    };
+
+    this.onPlayerListUpdate = this.onPlayerListUpdate.bind(this);
+    this.onGameStart = this.onGameStart.bind(this);
+    this.startGame = this.startGame.bind(this);
+  }
+
+  async componentDidMount() {
+    //Grab gameId & timeLimit to set those as well as gameName
```

Comment:
nit: add space between // and start of comment. Not really sure why it's standard coding convention but it is